src/Hint/Context.hs

```diff
@@ -50,9 +56,27 @@ newPhantomModule =
        let nums = concat [show (abs n::Int), show p, filter isDigit $ concat (ls ++ is)]
        let mod_name = 'M':nums
        --
-       tmp_dir <- liftIO getTemporaryDirectory
+       tmp_dir <- getPhantomDirectory
        --
-       return PhantomModule{pmName = mod_name, pmFile = tmp_dir </> nums}
+       return PhantomModule{pmName = mod_name, pmFile = tmp_dir </> mod_name <.> "hs"}
```

Comment:
Note also that the file name here is changed unconditionally. They are temporary files so nobody *should* care, but it may break a few people's cleanup scripts. I'm inclined to leave it unconditional though.